packages/flame/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,179 @@
+  TapUpInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    TapUpDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class LongPressStartInfo extends PositionInfo<LongPressStartDetails> {
+  LongPressStartInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    LongPressStartDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class LongPressEndInfo extends PositionInfo<LongPressEndDetails> {
+  final Vector2 velocity;
+
+  LongPressEndInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    this.velocity,
+    LongPressEndDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class LongPressMoveUpdateInfo extends PositionInfo<LongPressMoveUpdateDetails> {
+  LongPressMoveUpdateInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    LongPressMoveUpdateDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class ForcePressInfo extends PositionInfo<ForcePressDetails> {
+  final double pressure;
+
+  ForcePressInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    this.pressure,
+    ForcePressDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+// TODO(erick) This class, and the next one has a bunch of info, we need to check the ones
+// that could wrapped directly by the info class
+class PointerScrollInfo extends BaseInfo<PointerScrollEvent> {
+  final Vector2 position;
+  final Vector2 gameWidgetPosition;
+  final Vector2 scrollDelta;
+
+  PointerScrollInfo(
+    this.position,
+    this.gameWidgetPosition,
+    this.scrollDelta,
+    PointerScrollEvent raw,
+  ) : super(raw);
+}
+
+class PointerHoverInfo extends BaseInfo<PointerHoverEvent> {
+  final Vector2 position;
+  final Vector2 gameWidgetPosition;
+
+  PointerHoverInfo(
+    this.position,
+    this.gameWidgetPosition,
+    PointerHoverEvent raw,
+  ) : super(raw);
+}
+
+class DragDownInfo extends PositionInfo<DragDownDetails> {
+  DragDownInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    DragDownDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class DragStartInfo extends PositionInfo<DragStartDetails> {
+  DragStartInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    DragStartDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class DragUpdateInfo extends PositionInfo<DragUpdateDetails> {
+  final Vector2 delta;
+  DragUpdateInfo(
+    Vector2 position,
+    Vector2 gameWidgetPosition,
+    Vector2 globalPosition,
+    this.delta,
+    DragUpdateDetails raw,
+  ) : super(position, gameWidgetPosition, globalPosition, raw);
+}
+
+class DragEndInfo {
```

Comment:
my idea of adding BaseInfo was that this should extent it right? (same below a few times)

Reply:
Yeah, I totally forgot about these last classes, there are so many, these slipped my attention.

Changed